src/components/NotesView.vue

```diff
@@ -0,0 +1,270 @@
+<template>
+	<NcAppContent pane-config-key="note" :show-details="showNote" @update:showDetails="hideNote">
+		<template slot="list">
+			<NcAppContentList class="content-list">
+				<div class="content-list__search">
+					<input type="search" :placeholder="t('notes', 'Search for notes')" @input="event => searchText = event.target.value">
+				</div>
+
+				<NotesList v-if="groupedNotes.length === 1"
+					:notes="groupedNotes[0].notes"
+					@note-selected="onNoteSelected"
+				/>
+				<template v-for="(group, idx) in groupedNotes" v-else>
+					<NotesCaption v-if="group.category && category!==group.category"
+						:key="group.category"
+						:title="categoryToLabel(group.category)"
+					/>
+					<NotesCaption v-if="group.timeslot"
+						:key="group.timeslot"
+						:title="group.timeslot"
+					/>
+					<NotesList
+						:key="idx"
+						:notes="group.notes"
+						@note-selected="onNoteSelected"
+					/>
+				</template>
+				<div
+					v-if="displayedNotes.length != filteredNotes.length"
+					v-observe-visibility="onEndOfNotes"
+					class="loading-label"
+				>
+					{{ t('notes', 'Loading …') }}
+				</div>
+				<div v-if="getFilteredTotalCount > 0" class="content-list__search-more">
+					<NcButton @click="onCategorySelected(null)">
+						{{ t('notes', 'Find in all categories') }}
+					</NcButton>
+				</div>
+			</NcAppContentList>
+		</template>
+
+		<NcAppContentDetails>
+			<Note v-if="showNote" :note-id="noteId" @note-deleted="onNoteDeleted" />
+		</NcAppContentDetails>
+	</NcAppContent>
+</template>
+
+<script>
+
+import {
+	NcAppContent,
+	NcAppContentList,
+	NcAppContentDetails,
+	NcButton,
+} from '@nextcloud/vue'
+import { categoryLabel } from '../Util.js'
+import NotesList from './NotesList.vue'
+import NotesCaption from './NotesCaption.vue'
+import store from '../store.js'
+import Note from './Note.vue'
+
+import { ObserveVisibility } from 'vue-observe-visibility'
+
+export default {
+	name: 'NotesView',
+
+	components: {
+		NcAppContent,
+		NcAppContentList,
+		NcAppContentDetails,
+		NcButton,
+		Note,
+		NotesList,
+		NotesCaption,
+	},
+
+	directives: {
+		'observe-visibility': ObserveVisibility,
+	},
+
+	props: {
+		noteId: {
+			type: String,
+			required: true,
+		},
+	},
+
+	data() {
+		return {
+			timeslots: [],
+			monthFormat: new Intl.DateTimeFormat(OC.getLanguage(), { month: 'long', year: 'numeric' }),
+			lastYear: new Date(new Date().getFullYear() - 1, 0),
+			showFirstNotesOnly: true,
+			showNote: true,
+		}
+	},
+
+	computed: {
+		getFilteredTotalCount() {
+			return store.getters.getFilteredTotalCount()
+		},
+
+		category() {
+			return store.getters.getSelectedCategory()
+		},
+
+		filteredNotes() {
+			return store.getters.getFilteredNotes()
+		},
+
+		displayedNotes() {
+			if (this.filteredNotes.length > 40 && this.showFirstNotesOnly) {
```

Comment:
Why you catch for `> 40` but slice to `0-30` ?

Reply:
Something to consider separately I'd say as it is not introduced here but originating from https://github.com/nextcloud/notes/pull/424/files#diff-84c733529b3b3fd6edfc369f85fe95b5cfb8d1f0a74d39712293aa3b037a0848R113-R117 s